addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -379,3 +382,255 @@ def _l10n_es_edi_facturae_render_facturae(self):
     def _get_edi_doc_attachments_to_export(self):
         # EXTENDS 'account'
         return super()._get_edi_doc_attachments_to_export() + self.l10n_es_edi_facturae_xml_id
+
+    # -------------------------------------------------------------------------
+    # IMPORT
+    # -------------------------------------------------------------------------
+
+    def _get_edi_decoder(self, file_data, new=False):
+        def is_facturae(tree):
+            return tree.tag == '{http://www.facturae.es/Facturae/2014/v3.2.1/Facturae}Facturae'
+
+        if file_data['type'] == 'xml' and is_facturae(file_data['xml_tree']):
+            return self._import_invoice_facturae
+
+        return super()._get_edi_decoder(file_data, new=new)
+
+    def _import_invoice_facturae(self, invoice, file_data, new=False):
+        tree = file_data['xml_tree']
+        is_bill = invoice.move_type.startswith('in')
+        partner = self._import_get_partner(tree, is_bill)
+        self._import_invoice_facturae_invoices(invoice, partner, tree)
+
+    def _import_get_partner(self, tree, is_bill):
+        # If we're dealing with a vendor bill, then the partner is the seller party, if an invoice then it's the buyer.
+        party = tree.xpath('//SellerParty/*[2]') if is_bill else tree.xpath('//BuyerParty/*[2]')
+        if party:
+            partner_vals = self._import_extract_partner_values(party[0])
+            return self._import_create_or_retrieve_partner(partner_vals)
+        return None
+
+    def _import_extract_partner_values(self, party_node):
+        name = find_xml_value('.//CorporateName|.//Name', party_node)
+        first_surname = find_xml_value('.//FirstSurname', party_node)
+        second_surname = find_xml_value('.//SecondSurname', party_node)
+        phone = find_xml_value('.//Telephone', party_node)
+        mail = find_xml_value('.//ElectronicMail', party_node)
+        country_code = find_xml_value('.//CountryCode', party_node)
+        vat = find_xml_value('.//TaxIdentificationNumber', party_node)
+
+        full_name_parts = [part for part in [name, first_surname, second_surname] if part is not None]
+        full_name = ' '.join(full_name_parts)
+
+        return {'name': full_name, 'vat': vat, 'phone': phone, 'email': mail, 'country_code': country_code}
+
+    def _import_create_or_retrieve_partner(self, partner_vals):
+        name = partner_vals['name']
+        vat = partner_vals['vat']
+        phone = partner_vals['phone']
+        mail = partner_vals['email']
+        country_code = partner_vals['country_code']
+
+        partner = self.env['res.partner']._retrieve_partner(name=name, vat=vat, phone=phone, mail=mail)
+
+        if not partner and name:
+            partner_vals = {'name': name, 'email': mail, 'phone': phone}
+            country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else False
```

Comment:
> Don't we already include all countries in base?

People could delete and recreate for instance. Also, some countries don't recognize the same countries as others. And there can always be geopolitical changes.

> What do you mean by "do a query for the `.exists()`

People often think that `ref` is better than `search` because it makes less round trips with the SQL server thanks to caching, but it is not true (however it is true that we are making a non indexed SELECT versus an indexed one, on a small table)